### PR TITLE
Workflows permissions and transitions

### LINKED ISSUE #92: Workflows permissions [epic?]: define new roles, states, transitions and permissions.
## User story

**As a:** product owner
**I want:** publishing workflow roles and permissions to be enabled by default when I enable to localgov_workflows
**So that:** we have a sensible starting point for publishing workflows.

## Notes: 

These can always be changed once installed, this is meant as a starting point for councils that want to delegate authoring permissions to a wider set of users while retaining review and publishing permisisons to a few.

## Acceptance tests: 

See 
 - [Miro board](https://miro.com/app/board/o9J_lHm9M2s=/)
 - [Permissions matrix](https://docs.google.com/spreadsheets/d/1EXkhl0K1YRpbWxbPDMQOudT9Z6CSpf9dpXsl9XJwHXs/edit#gid=0)

From these documents, we have the following: 

Contributor [contributor]:
- can write content (including creating paragraph components, upload media etc)
- reuse paragraphs from the paragraph library
- cannot publish
- sends content to approval by an Editor
- typically a service team member

Author [author]:

- has elevated access rights
- can write content
- can publish their own content
- cannot publish content sent for approval
- typically a digitally savvy service team member or someone in comms
- can (optionally) send their contents for peer review

Editor [editor]:

- can write and publish content
- can approve or reject edits
- access to an approvals dashboard
- typically members of central digital team
- can edit menus, headers, footers (and other areas which don't have workflow)

We also noted that we need a user manager. 

User manager: 

- can add users
- can assign certain roles (contributor, author, editor) to users (but not assign the super user / administrator role)

## Notes

LocalGov Drupal with the localgov_workflows module currently defines the following roles: 

- Content designer (machine name: editor)
- Author

We will need to decide if the permissions will be kept in the [localgov_roles module](https://github.com/localgovdrupal/localgov_core/tree/2.x/modules/localgov_roles) or [localgov_workflows](https://github.com/localgovdrupal/localgov_core/tree/2.x/modules/localgov_workflows), or in specific feature modules. 

## Tasks

- [x] Add new roles 
- [x] Ensure correct workflow states are in place
- [x] Configure permissions
- [x] Write phpunit tests for each role
- [x] Enable translations to test revisions / workflows compatibility ( not planning to ship with this, just to make sure we catch any architectural issues. )
- [x] Ensure the default workflows are enabled on all content types. 
- [ ] Install permissions in modules for all content types.

### PR DESCRIPTION
**Pull request by stephen-cox**
Adds contributor and user manager roles to localgov_roles and updates default permissions assigned to all roles in localgov_workflows.

Closes #92

### CODE REVIEW

**Comment by stephen-cox** [comment]
@willguv @msayoung This work has been deployed to the test site: https://test.localgovdrupal.org/

I have created users with each of the redefined roles: LocalGov Editor, LocalGov Author, LocalGov Contributor, and LocalGov User Manager.

To aid testing I have installed the [Masquerade](https://www.drupal.org/project/masquerade) module which will allow you to easily switch between each of these users while testing.

If you login as the LocalGov Admin user (password pinned in the Slack #technical-group channel), you'll see the masquerade block in the footer so you can switch users.

**Comment by stephen-cox** [comment]
The functionality of the work hear has been tested and reviewed on #92 so this is ready for a code review; which is mainly tests.

File: modules/localgov_workflows/tests/src/Functional/WorkflowsTranslationTest.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    // Archive English page.
+    $this->drupalGet('node/' . $en_node->id() . '/edit');
+    $this->submitForm([
+      'moderation_state[0][state]' => 'archived',
+    ], 'Save (this translation)');
+    $node_storage->resetCache([$en_node->id()]);
+    $en_node = $node_storage->loadRevision($node_storage->getLatestRevisionId($en_node->id()));
+    $cy_translation = $en_node->getTranslation('cy');
+    $this->assertEquals('archived', $en_node->moderation_state->value);
+    $this->assertEquals('published', $cy_translation->moderation_state->value);
```
Comment (finnlewis) [inline modules/localgov_workflows/tests/src/Functional/WorkflowsTranslationTest.php:115]: I like this testing of translations @stephen-cox,  protecting the future!

**Review by finnlewis — APPROVED** [review]
I'm happy with this. I've tested it locally and read the config and code.
Would be good to get an OK from @andybroomfield or @Adnan-cds  - based on the discussion on  https://github.com/localgovdrupal/localgov_core/issues/111 we plan to get this work in first, then split off the localgov_workflows as a separate module.

**Review by andybroomfield — APPROVED** [review]
Ok as per #111 this is split out when localgov_workflow is made its own module.

**Pull request merged**